Common/Src/assert.c

```diff
@@ -0,0 +1,10 @@
+#include "assert.h"
+#include <stdbool.h>
+
+void zp_assert(bool expr) {
+    if (expr) {
+#warning This should call the los hardfault handler, or something more appropriate
+        while (1) {
+        }
+    }
```

Comment:
Add a stack trace printout so that we know where the assert was called, as well as printing out the location of where the assert failed to aid debugging